[CDAP-14553] [CDAP-14599] Adds generic plugin function widget

ajainarayanan opened the pull request:
**Feature**
- Adds a generic plugin function widget to make a request to backend and update a specific property in the plugin.
- This is different than schema updates which needs to be broadcasted.

**Bugfixes**
- Fixes filer in control center to properly update store.
- Fixes radio-group widget to have proper unique html ids.
- Removes replace:dev from gulp watch. This will result in caching during development. A workaround is to disable cache in browser we are developing.
With replace:dev dist folder was exploding and the html files were not properly updated with changed script tags. 
This change just simplifies for dev workflow. The distribute will still have all hashes added to files.

**Note**:
Docs for the plugin function widgets are still pending. Will either add to this PR or will create a separate PR for it.

**JIRA**: 
https://issues.cask.co/browse/CDAP-14553
https://issues.cask.co/browse/CDAP-14599

**Build**:
https://builds.cask.co/browse/CDAP-UDUT155

**Docs:**
https://builds.cask.co/artifact/CDAP-DQB450/shared/build-1/Docs-HTML/html/developer-manual/pipelines/developing-plugins/presentation-plugins.html#H3270

elfenheart commented on cdap-ui/app/directives/plugin-functions/functions/get-property-value/get-property-value.html:
```diff
@@ -0,0 +1,19 @@
+-->
+
+<button class="btn btn-default pull-right" type="button" ng-click="GetPropertyValueController.openModal()">
+  Get Value
```
should we make this button text configurable?

elfenheart commented on cdap-ui/app/directives/plugin-functions/functions/get-property-value/get-property-value-modal.html:
```diff
@@ -0,0 +1,49 @@
+
+  <fieldset disabled="true" ng-if="GetPropertyValueModalController.propertyValue">
+    <my-json-textbox
+      ng-model="GetPropertyValueModalController.propertyValue"
```
we are showing the value, but should we show what property will actually be modified if we hit apply?

ajainarayanan replied:
So the design was the plugin developer will add this plugin function to the property for which he/she wants to get the value from backend. Say if I want to get the path property in file plugin I would add the plugin function in widget json for the path property.

elfenheart commented on cdap-ui/app/directives/plugin-functions/functions/get-property-value/get-property-value.js:
```diff
@@ -0,0 +1,107 @@
+        var fnConfig = $scope.fnConfig;
+        var outputProperty = fnConfig['output-property'];
+        var methodName = fnConfig['plugin-method'] || 'getSchema';
+        vm.buttonlabel = myHelpers.objectQuery(fnConfig, 'widget-attributes', 'buttonlabel') || 'Get Value';
```
nit: camel case or dash separate the words?

elfenheart reviewed (approved):
minor comment, otherwise 👍 LGTM

elfenheart reviewed (approved):
👍 LGTM

ajainarayanan committed:
[CDAP-14553] [CDAP-14599] Adds generic plugin function widget + fixes the filter in Control Center home page

  - Removes the usage of output-property from the widget json spec. It is no longer being used in the plugin functions
    outputSchema widget updates only the schema (which broadcasts it in the node) and getPropertyValue is configured
    to the property name that needs to fetch the value from the backend. This takes the necessity of having the
    output-property in the widget json

  - Updates plugin nodeconfig to pass in node config to plugin-functions directive
  - Docs for plugin functions in widget json
  - Updates radio-group widget to assign valid html ids for radio buttons
  - Fixes filter in EntityListView to properly show filtered cdap artifacts
  - Fixes modal story to use proper bootstrap 4 css clas
  - Replaces the use of rev:replace in gulp watch build. This was causing new files
    to be generated on each watch update. This becomes worse when we don't update
    the html files with new script tags.

    This forced us to delete the dist directory and restart the build as the newly generated
    script was not updated in the html

    Reverting this will always generate/update the same script added to the base html document.
    During development we need to disable cache to avoid running into issues.

elfenheart reviewed (approved):
👍 LGTM

ajainarayanan commented:
Latest build: https://builds.cask.co/browse/CDAP-UDUT155-6
